src/openloco/ui.cpp

```diff
@@ -760,23 +767,116 @@ namespace openloco::ui
         SDL_ShowSimpleMessageBox(SDL_MESSAGEBOX_BUTTON_RETURNKEY_DEFAULT, title.c_str(), message.c_str(), window);
     }
 
-    bool updateSDLResolution()
+    static config::resolution_t getDisplayResolutionByMode(config::screen_mode mode)
     {
-        SDL_DisplayMode mode;
-        SDL_GetWindowDisplayMode(window, &mode);
+        config::resolution_t defaultResolution = { 640, 480 };
+        config::resolution_t* outputResolution = &defaultResolution;
 
-        auto& cfg = config::get();
-        mode.h = cfg.resolution_height;
-        mode.w = cfg.resolution_width;
+        auto& config = config::get_new();
+        switch (mode)
+        {
+            case config::screen_mode::window:
+                if (config.display.window_resolution.isPositive())
+                    outputResolution = &config.display.window_resolution;
+                break;
+
+            case config::screen_mode::fullscreen:
+                if (config.display.fullscreen_resolution.isPositive())
+                    outputResolution = &config.display.fullscreen_resolution;
+                break;
+
+            case config::screen_mode::fullscreen_borderless:
+                break;
+
+            default:
+                console::error("Unrecongised display mode");
+                break;
```

Comment:
Can be removed, as it shouldn't compile due to the compiler type checking `config::screen_mode`.